src/tracing/eBPF/CMakeLists.txt

```diff
@@ -0,0 +1,168 @@
+build_libbpf()
+build_bpftool()
+
+# libbpf requires zlib
+find_package(ZLIB REQUIRED)
+
+# Paths
+set(EBPF_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+set(EBPF_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
+
+# Include paths for BPF compilation
+set(EBPF_INCLUDES
+  -I${EBPF_SOURCE_DIR}
+  -I${EBPF_BINARY_DIR}
+  -I${LIBBPF_UAPI_INCLUDE_DIR}
+  -I${LIBBPF_SRC_DIR})
+
+# Get clang BPF system includes
+execute_process(
+  COMMAND ${CLANG_EXECUTABLE} -v -E -
+  INPUT_FILE /dev/null
+  ERROR_VARIABLE CLANG_STDERR
+  OUTPUT_QUIET
+)
+string(REGEX MATCHALL "[ \t]([^\n]+)" CLANG_INCLUDE_DIRS "${CLANG_STDERR}")
+set(CLANG_BPF_SYS_INCLUDES "")
+set(IN_SEARCH_LIST FALSE)
+foreach(LINE ${CLANG_INCLUDE_DIRS})
+  string(STRIP "${LINE}" LINE)
+  if(LINE MATCHES "^#include <...> search starts here:")
+    set(IN_SEARCH_LIST TRUE)
+  elseif(LINE MATCHES "^End of search list")
+    set(IN_SEARCH_LIST FALSE)
+  elseif(IN_SEARCH_LIST AND LINE MATCHES "^/")
+    list(APPEND CLANG_BPF_SYS_INCLUDES "-idirafter${LINE}")
+  endif()
+endforeach()
+
+# Generate bpf_ceph_exported.h from source headers
+add_custom_command(
+  OUTPUT ${EBPF_BINARY_DIR}/bpf_ceph_exported.h
+  COMMAND ${EBPF_SOURCE_DIR}/gen_bpf_types.sh ${EBPF_BINARY_DIR}/bpf_ceph_exported.h
+  DEPENDS
+    ${CMAKE_SOURCE_DIR}/src/include/ceph_fs.h
+    ${CMAKE_SOURCE_DIR}/src/msg/Message.h
+    ${CMAKE_SOURCE_DIR}/src/include/rados.h
+    ${CMAKE_SOURCE_DIR}/src/os/bluestore/BlueStore.h
+    ${EBPF_SOURCE_DIR}/gen_bpf_types.sh
+    ${EBPF_SOURCE_DIR}/bpf_ceph_types.h
+  COMMENT "Generating bpf_ceph_exported.h from Ceph headers"
+  VERBATIM
+)
+
+add_custom_target(bpf_ceph_export_target
+  DEPENDS ${EBPF_BINARY_DIR}/bpf_ceph_exported.h
+)
+
+# Build BPF object file
+add_custom_command(
+  OUTPUT ${EBPF_BINARY_DIR}/radostrace.bpf.o
+  COMMAND ${CLANG_EXECUTABLE} -g -O2 -target bpf
+    -D__TARGET_ARCH_${ARCH}
+    ${EBPF_INCLUDES}
+    ${CLANG_BPF_SYS_INCLUDES}
+    -c ${EBPF_SOURCE_DIR}/radostrace.bpf.c
+    -o ${EBPF_BINARY_DIR}/radostrace.tmp.bpf.o
+  COMMAND ${BPFTOOL_EXECUTABLE} gen object
+    ${EBPF_BINARY_DIR}/radostrace.bpf.o
+    ${EBPF_BINARY_DIR}/radostrace.tmp.bpf.o
+  DEPENDS
+    ${EBPF_SOURCE_DIR}/radostrace.bpf.c
+    ${EBPF_BINARY_DIR}/bpf_ceph_exported.h
+    ${EBPF_SOURCE_DIR}/bpf_ceph_types.h
+    ${EBPF_SOURCE_DIR}/bpf_utils.h
+    bpftool_ext
+    libbpf_ext
+    bpf_ceph_export_target
+  COMMENT "Compiling BPF program radostrace.bpf.c"
+  VERBATIM
+)
+
+# Generate BPF skeleton
+add_custom_command(
+  OUTPUT ${EBPF_BINARY_DIR}/radostrace.skel.h
+  COMMAND ${BPFTOOL_EXECUTABLE} gen skeleton
+    ${EBPF_BINARY_DIR}/radostrace.bpf.o
+    > ${EBPF_BINARY_DIR}/radostrace.skel.h
+  DEPENDS ${EBPF_BINARY_DIR}/radostrace.bpf.o
+  COMMENT "Generating BPF skeleton radostrace.skel.h"
+  VERBATIM
+)
+
+# Build dwarf_parser object
+add_library(dwarf_parser OBJECT
+  ${EBPF_SOURCE_DIR}/dwarf_parser.cc
+  ${EBPF_SOURCE_DIR}/dwarf_parser.h
+)
+
+target_include_directories(dwarf_parser PRIVATE
```

Comment:
if `dwarf_parser` source files includes libbpf headers, and links against libbpf libraries, it should `link` against libbpf like:
```cmake
target_link_libraries(dwarf_parser
  libbpf::libbpf)
```

CMake is able to pull the libraries and header include directories in, when building `dwarf_parser` and when linking `dwarf_parser` and its consumers.